### DISCUSSION TITLE
Method to read event status?

### DISCUSSION THREAD

**Discussion by WalkingDot** [roleoroleo/yi-hack-Allwinner-v2#402] [Q&A] [answered]
Hello,
is there a recommended method to read camera events (something is triggered) without MQTT?
Currently I know only 2 ways, but none seems optimal to me...

1.) Kill and restart the process "/tmp/sd/yi-hack/bin/ipc_multiplexer" to get the output.

2.) Check if the temporary file exist "/tmp/sd/record/tmp.mp4.tmp".

I would be very happy about any help.
At the moment I am trying to write an extension for mail notifications and for this I would like to use the proper method.

Best regards
WalkingDot

**Comment by WalkingDot** [comment, marked as answer]
Currently the mail function can be started manually via ssh. Right now it's running in a 5 minute test-loop and successfully sends smtp-mails, with attached low quality snapshots.
Just a starter/trigger, user-friendly smtp-login parameters and some small improvements are missing... then I could post a test version.

**Reply by rpkilgore** [reply]
I'd be interested in testing this extension.  Is it out here on git hub?

**Reply by WalkingDot** [reply]
@rpkilgore you are welcome. At the moment I try to reduce allocations and memory usage. If this runs some more efficient, I'll put the code online and then gladly let you know. In the meantime I still hope that roleoroleo might know a proper method to read out the event status somewhere or maybe execute the binary on motion/sound events.

**Reply by roleoroleo** [reply]
I created ipc_multiplexer to share the events with custom services.
It reads from ipc_dispatch queue and repeats the event to ipc_dispatch_x (1 to 9).
ipc_dispatch_1 is used by mqttv4.
You should read ipc_dispatch_2 (or following). ipc_read is already available as a binary in the hack.
Otherwise you can create your own program using ipc_read as a template:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/blob/master/src/ipc_cmd/ipc_cmd/ipc_read.c

**Reply by WalkingDot** [reply]
Hi roleo, many thanks for your explanation! This helps me a lot.
I think I will use your ipc_read, with an optional argument to choose the dispatch number. 
Now I feel kind of stupid, because I didn't recognize this before ;-)

**Reply by WalkingDot** [reply]
@rpkilgore if you are still interested. Today I have uploaded the first test version.
https://github.com/WalkingDot/yihmail

**Reply by roleoroleo** [reply]
Very nice.

**Reply by WalkingDot** [reply]
thanks, once the testing is finished, feel free to take parts or the binary. I could also create a lighter version without event daemon, log etc...

**Reply by rpkilgore** [reply]
Very nice, thanks for sharing!
